### ISSUE TITLE
[BUG] prefer-immediate-return does not work inside nested blocks

### ISSUE THREAD

**Issue by furaiev** [dart-code-checker/dart-code-metrics#754]
E.g. 

```
void calculateSum(int a, int b) {
    final sum = a + b;

    return sum; // LINT
}

void calculateSum(int a, int b) {
    try {
        final sum = a + b;

        return sum; // LINT DOES NOT WORK
    } ...
}
```

**Comment by dkrutskikh** [comment]
@roman-petrov please take a look

**Comment by roman-petrov** [comment]
@furaiev Thank you, good catch! This seems to be clear so I will make a PR in a few days.

**Comment by incendial** [comment]
Available in 4.14.0 🚀
